src/bitmessagemain.py

```diff
@@ -1,49 +1,55 @@
 #!/usr/bin/python2.7
-# Copyright (c) 2012-2016 Jonathan Warren
-# Copyright (c) 2012-2018 The Bitmessage developers
-# Distributed under the MIT/X11 software license. See the accompanying
-# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+# pylint: disable=no-self-use,too-many-branches,too-many-statements,too-many-locals
+"""
+bitmessagemain.py
+=================
 
-# Right now, PyBitmessage only support connecting to stream 1. It doesn't
-# yet contain logic to expand into further streams.
+Copyright (c) 2012-2016 Jonathan Warren
+Copyright (c) 2012-2018 The Bitmessage developers
+Distributed under the MIT/X11 software license. See the accompanying
+file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-# The software version variable is now held in shared.py
+Right now, PyBitmessage only support connecting to stream 1. It doesn't
+yet contain logic to expand into further streams.
 
+The software version variable is now held in shared.py
+
+"""
+
+from __future__ import absolute_import
+
+import ctypes
+import errno
+import getopt
 import os
+import signal
+import socket
 import sys
+import threading
+from random import randint
+from struct import pack
+from subprocess import call
+from time import sleep
 
 app_dir = os.path.dirname(os.path.abspath(__file__))
 os.chdir(app_dir)
 sys.path.insert(0, app_dir)
 
-
-import depends
-depends.check_dependencies()
-
 # Used to capture a Ctrl-C keypress so that Bitmessage can shutdown gracefully.
-import signal
 # The next 3 are used for the API
 from singleinstance import singleinstance
-import errno
-import socket
-import ctypes
-from struct import pack
-from subprocess import call
-from time import sleep
-from random import randint
-import getopt
 
 from api import MySimpleXMLRPCRequestHandler, StoppableXMLRPCServer
 from helper_startup import (
     isOurOperatingSystemLimitedToHavingVeryFewHalfOpenConnections
 )
 
 import defaults
```

Comment:
Sorry, this PR is a work in progress, as per the one it replaced. I needed a PR# to reference in the old ticket. If I close this one I'll have to open another PR since I don't have permission to re-open in such cases. Just hold off reviewing this on until I mark it as ready for review. There's three other PRs #1263, #1264 and #1268 that need review though.

Reply:
@coffeedogs any update here?